signalr-go-server/pkg/signalr/hubprotocol.go

```diff
@@ -0,0 +1,62 @@
+package signalr
+
+import (
+	"bytes"
+	"io"
+)
+
+// HubProtocol interface
+type HubProtocol interface {
+	ReadMessage(buf *bytes.Buffer) (interface{}, error)
+	WriteMessage(message interface{}, writer io.Writer) error
+	UnmarshalArgument(argument interface{}, value interface{}) error
```

Comment:
This is a bit different in the other clients/servers and instead is another interface. Lets leave this for now to though.